code/segmentedtickerplant/stplog.q

```diff
@@ -52,55 +52,67 @@ logname[`custom]:{[dir;tab;p]
 ///////////////////////////////////////////////////////////////////////////////////////
 
 // Update and timer functions in three batch modes ////////////////////////////////////
+// preserve pre-existing definitions
+upd:@[value;`.stplg.upd;enlist[`]!enlist ()];
+zts:@[value;`.stplg.zts;enlist[`]!enlist ()];
 
-upd:zts:batch:enlist[`]!enlist ()
+// Number of update messages received for each table
+msgcount:rowcount:tmpmsgcount:tmprowcount:enlist[`]!enlist ()
+
+// Sequence number
+seqnum:0
+
+// Functions to add columns on updates
+updtab:@[value;`.stplg.updtab;enlist[`]!enlist {(enlist(count first x)#y),x}]
 
 // If set to autobatch, publish and write to disk will be run in batches
-upd[`autobatch]:{[t;x]
-  x:.stpps.updtab[t]@x;
-  @[`.stplg.batch;t;,;enlist x]
+// insert to table in memory, on a timer flush the table to disk and publish, update counts
+upd[`autobatch]:{[t;x;now]
+  t insert updtab[t] . (x;now);
  };
 
 zts[`autobatch]:{
   {[t]
-    x:batch[t];
-    if[count x;
-      .stpps.pub[t]'[x];
-      `..loghandles[t] (`upd;t),/:enlist each x]
+    if[count value t;
+      `..loghandles[t] enlist (`upd;t;flip value t);
```

Comment:
Looks like we're writing data to disk as lists for defaultbatch and immediate, but as a table for autobatch. We're publishing as tables for defaultbatch and autobatch, but lists for immediate. Would this cause any problems potentially?

Reply:
You are right - this should be 
value flip value t
So should then be the same (lists)